### ISSUE TITLE
Story S1.3: Configuration Management System

### ISSUE THREAD

**Issue by iudofia2026** [iudofia2026/NBA-Gambling-Addicts#8]
## Parent
Epic E1: Project Infrastructure & Setup (#1)

## Priority
P0 (Critical)

## Complexity
M (Medium)

## Description
Implement configuration management system using YAML files and environment variables to manage different settings for development, testing, and production environments. Enable easy switching between configurations without code changes.

## Acceptance Criteria
- [ ] Given config/config.yaml, when application loads, then correct environment-specific settings are used
- [ ] Given .env file, when accessing sensitive data, then credentials are not hardcoded
- [ ] Given multiple environments, when switching configs, then no code changes required
- [ ] Given configuration module, when other components need settings, then they import from single source
- [ ] Given .env.example, when new developer sets up, then they know what environment variables to configure

## Integration Notes
**Configuration Structure:**
- config/config.yaml: Main configuration file with environment-specific sections
- .env: Local environment variables (gitignored)
- .env.example: Template for required environment variables
- src/utils/config.py: Configuration loader module

**Key Configuration Areas:**
- Data paths (raw, processed, model output)
- Model hyperparameters (default values)
- Feature engineering parameters (rolling window sizes, etc.)
- Kaggle API credentials (from .env)
- Logging configuration
- Random seeds for reproducibility

**Example config.yaml structure:**
```yaml
data:
  raw_data_path: "data/raw"
  processed_data_path: "data/processed"
  models_path: "models"

features:
  rolling_windows: [3, 10]
  min_games_played: 5

models:
  random_forest:
    n_estimators: 100
    max_depth: 10
    random_state: 42
  xgboost:
    max_depth: 6
    learning_rate: 0.1
    n_estimators: 100
    random_state: 42

logging:
  level: "INFO"
  format: "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
```

## Technical Considerations
- Use pyyaml for config file parsing
- Use python-dotenv for .env file loading
- Create singleton Config class for application-wide access
- Validate configuration on load (check paths exist, required fields present)
- Support environment variable overrides (e.g., DATA_PATH env var overrides config.yaml)

## Labels
story, infrastructure, configuration, P0

**Comment by iudofia2026** [comment]
Closing all issues as requested
